src/common/calendar/TimePickerCalendar.js

```diff
@@ -180,9 +180,22 @@ class TimePickerCalendar extends Component {
           NOTIFICATION_TYPE.INFO, t('TimePickerCalendar.info.shouldBeReservedWholeDayText'),
         );
       }
+
       const startMoment = moment(selected.start).toJSON();
       const selectedDate = startMoment.split('T')[0];
-      const slots = this.getSlotsForDate(selectedDate, resource);
+
+      // select slots from next exact hour
+      // e.g. if time now 11:40, next available slot should be 12:00
+      // even if exact hour, add one hour to prevent server validation rejecting
+      // time earlier than current
+      const minTime = moment().add(1, 'hour').startOf('hour');
```

Comment:
I think this logic makes sense here. Generally the smallest possible time slot for reservations is 30 mins (this is configurable on the resource) but since this only affects whole day reservations, I don't think we need to take that into consideration.